daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -718,3 +741,196 @@ class UnparseResult(dp: DataProcessor, ustate: UState)
+      case expectedUniqueURNRegex(_, bp) if bp == "BlobPrefix" => saxBlobPrefixProperty
+      case expectedUniqueURNRegex(_, bs) if bs == "BlobSuffix" => saxBlobSuffixProperty
+      case _ =>
+        throw new SAXNotRecognizedException("Property unsupported: " + name + "." +
+          "Supported properties are: " + saxSupportedProperties.mkString(", "))
+    }
+    prop
+  }
+
+  override def setProperty(name: String, value: AnyRef): Unit = {
+    val expectedUniqueURNRegex = s"(${XMLUtils.DAFFODIL_SAX_URN_ROOT}:)(\\w+)".r
+    try {
+      name match {
+        case expectedUniqueURNRegex(_, pr) if pr == "ParseResult" => saxParseResultProperty = value
+          .asInstanceOf[ParseResult]
+        case expectedUniqueURNRegex(_, bd) if bd == "BlobDirectory" => saxBlobDirectoryProperty =
+          value.asInstanceOf[Path]
+        case expectedUniqueURNRegex(_, bp) if bp == "BlobPrefix" => saxBlobPrefixProperty = value
+          .asInstanceOf[String]
+        case expectedUniqueURNRegex(_, bs) if bs == "BlobSuffix" => saxBlobSuffixProperty = value
+          .asInstanceOf[String]
+        case _ =>
+          throw new SAXNotRecognizedException("Property unsupported: " + name + "." +
+            "Supported properties are: " + saxSupportedProperties.mkString(", "))
+      }
+    } catch {
+      case e: ClassCastException =>
+        throw new SAXNotSupportedException("Unsupported value for property: " + name + "." +
+        "Supported properties/values types are: " + saxSupportedProperties.map{
+          p => s"$p: ${p.getClass}"}.mkString(", "))
+    }
+  }
+
+  override def setEntityResolver(resolver: EntityResolver): Unit = {
+    entityResolver = resolver
+  }
+
+  override def getEntityResolver: EntityResolver = entityResolver
+
+  override def setDTDHandler(handler: DTDHandler): Unit = {
+    dtdHandler = handler
+  }
+
+  override def getDTDHandler: DTDHandler = dtdHandler
+
+  override def setContentHandler(handler: ContentHandler): Unit = {
+    contentHandler = handler;
+  }
+
+  override def getContentHandler: ContentHandler = contentHandler
+
+  override def setErrorHandler(handler: ErrorHandler): Unit = {
+    errorHandler = handler;
+  }
+
+  override def getErrorHandler: ErrorHandler = errorHandler
+
+  override def parse(input: InputSource): Unit = {
+    val is = input.getByteStream
+    if(is != null) {
+      val isdis = InputSourceDataInputStream(is)
+      parse(isdis)
+    } else {
+      throw new IOException("Inputsource must be backed by Inputstream")
+    }
+  }
+
+  override def parse(systemId: String): Unit = {
+    throw new IOException("SAX parsing of systemId is unsupported")
+  }
+
+  def parse(isdis: InputSourceDataInputStream): Unit = {
+    val sio = createSAXInfosetOutputter(this)
+    val pr = dp.parse(isdis, sio)
+    handleDiagnostics(pr)
+    setProperty(s"${XMLUtils.DAFFODIL_SAX_URN_ROOT}:ParseResult", pr)
```

Comment:
I'd just set the saxPropetyParserResult variable directly. No need to go through setProperty, and that way users don't have a way to change the ParseResult property. This property should be read only.